inception-active-learning/src/main/java/de/tudarmstadt/ukp/inception/active/learning/ActiveLearningServiceImpl.java

```diff
@@ -102,4 +114,339 @@ public ActiveLearningServiceImpl(DocumentService aDocumentService,
         return model.getFlattenedPredictions(aState.getDocument().getName(), aSelectedLayer,
             windowBegin, windowEnd, aJcas, true);
     }
+
+    @Override
+    public void putSessionActive(User aUser, Project aProject, boolean aSesscionActive)
+    {
+        ActiveLearningUserState state = getState(aUser.getUsername(), aProject);
+        synchronized (state) {
+            state.setSessionActive(aProject, aSesscionActive);
+        }
+    }
+
+    @Override
+    public boolean getSessionActive(User aUser, Project aProject)
+    {
+        ActiveLearningUserState state = getState(aUser.getUsername(), aProject);
+        boolean sessionActive;
+        synchronized (state) {
+            sessionActive = state.isSessionActive(aProject);
+        }
+        return sessionActive;
+    }
+
+    @Override
+    public void putHasUnseenRecommendation(User aUser, Project aProject,
+        boolean aHasUnseenRecommendation)
+    {
+        ActiveLearningUserState state = getState(aUser.getUsername(), aProject);
+        synchronized (state) {
+            state.setHasUnseenRecommendation(aProject, aHasUnseenRecommendation);
+        }
+    }
+
+    @Override
+    public boolean getHasUnseenRecommendation(User aUser, Project aProject)
+    {
+        ActiveLearningUserState state = getState(aUser.getUsername(), aProject);
+        boolean hasUnseenRecommendation;
+        synchronized (state) {
+            hasUnseenRecommendation = state.isHasUnseenRecommendation(aProject);
+        }
+        return hasUnseenRecommendation;
+    }
+
+    @Override
+    public void putHasSkippedRecommendation(User aUser, Project aProject,
+        boolean aHasSkippedRecommendation)
+    {
+        ActiveLearningUserState state = getState(aUser.getUsername(), aProject);
+        synchronized (state) {
+            state.setHasSkippedRecommendation(aProject, aHasSkippedRecommendation);
+        }
+    }
+
+    @Override
+    public boolean getHasSkippedRecommendation(User aUser, Project aProject)
+    {
+        ActiveLearningUserState state = getState(aUser.getUsername(), aProject);
+        boolean hasSkippedRecommendation;
+        synchronized (state) {
+            hasSkippedRecommendation = state.isHasSkippedRecommendation(aProject);
+        }
+        return hasSkippedRecommendation;
+    }
+
+    @Override
+    public void putDoExistRecommender(User aUser, Project aProject, boolean aDoExistRecommenders)
```

Comment:
"set" methods start with `set` - applies to other methods as well.

Why is there a `doExistRecommender` flag in the state? Shouldn't this information be obtained from the recommender service? In the state, this information could easily be outdated.